uninstall.sh

```diff
@@ -0,0 +1,388 @@
+#!/bin/bash
+set -u
+shopt -s extglob
+
+abort() {
+  printf "%s\n" "$1"
+  exit 1
+}
+
+strip_s() {
+  local s
+  for s in "$@"; do
+    s=${s## }
+    echo "${s%% }"
+  done
+}
+
+dir_children() {
+  local p
+  for p in "$@"; do
+    [[ -d $p ]] || continue
+    find "$p" -mindepth 1 -maxdepth 1
+  done
+}
+
+# Set up temp dir
+tmpdir=/tmp/uninstall.$$
+mkdir -p "$tmpdir" || abort "Unable to create temp dir '$tmpdir'"
+trap '
+  rm -fr "$tmpdir"
+  # Invalidate sudo timestamp before exiting
+  /usr/bin/sudo -k
+' EXIT
+
+# Default options
+opt_force=""
+opt_quiet=""
+opt_dry_run=""
+opt_skip_cache_and_logs=""
+
+# global status to indicate whether there is anything wrong.
+failed=false
+
+un=$(uname)
+case "$un" in
+  Linux)
+    ostype=linux
+    homebrew_prefix_default=/home/linuxbrew/.linuxbrew
+  ;;
+  Darwin)
+    ostype=macos
+    homebrew_prefix_default=/usr/local
+    # Stock macOS doesn't have a realpath, so we make do with builtin Python
+    realpath() {
+      /usr/bin/python -c 'import os,sys;print(os.path.realpath(sys.argv[1]))' "$1"
```

Comment:
I'm afraid introducing dependency on _builtin Python_ is not a good option. Translation to Bash started from the deprecation of scripting languages (see [49764202](https://developer.apple.com/documentation/macos_release_notes/macos_catalina_10_15_release_notes)). I think Python will be installed while installing the command line tools but is it guaranteed to keep it after uninstallation? @MikeMcQuaid What do you think?

Reply:
> [49764202](https://developer.apple.com/documentation/macos_release_notes/macos_catalina_10_15_release_notes))

Oh. That's not good. Will find a pure-bash replacement.

Reply:
agreed with @itchyny. This is the method I tend to use: https://stackoverflow.com/a/35996728/5355

Reply:
OK, here's the version I'm inclined to go with. It takes care not to output anything except the canonicalized pathname, or nothing at all if it can't be canonicalized for whatever reason, especially link chains 100 or more levels deep (usually a symlink cycle):
```sh
realpath() {
  local dir base newpath
  dir=$(dirname "$1")
  base=$(basename "$1")
  (
    cd -P "$dir" &> /dev/null || exit 1
    redirs=0
    while [[ -L $base && $((redirs++)) -lt 100 ]]; do
      newpath=$(readlink "$base" 2>/dev/null) || exit 1
      dir=$(dirname "$newpath")
      base=$(basename "$newpath")
      cd -P "$dir" &> /dev/null || exit 1
    done
    if [[ -L $base ]]; then
      exit 1
    else
      echo "$(pwd -P)/$base"
    fi
  )
}
```
Comments?

Reply:
That seems excessively complicated. `cd` and `pwd` with an argument should be sufficient.

Reply:
That would only work if the entire path points to an existing directory, so it'll fail when we test, say, `$HOMEBREW_PREFIX/bin/brew`, as is already done in the uninstaller.

Reply:
If it doesn't exist it can/will return an empty string which can itself be filtered out.

Reply:
Are you talking about:
```
cd "$(dirname "$target")" && echo "$(pwd -P)/$(basename "$target")"
```
? I guess that might work.

Reply:
Aye, thanks!